packages/radio-player/src/radio-player.ts

```diff
@@ -0,0 +1,738 @@
+    return html`
+      <search-results-switcher
+        class="${this.shouldShowSearchResultSwitcher ? '' : 'hidden'}"
+        @searchResultIndexChanged=${this.searchResultIndexChanged}
+      >
+      </search-results-switcher>
+    `;
+  }
+
+  private get noSearchResultsTemplate(): TemplateResult {
+    return html`
+      <div
+        class="no-search-results-message ${this.shouldShowNoSearchResultMessage ? '' : 'hidden'}"
+      >
+        No search results.
+      </div>
+    `;
+  }
+
+  private get quickSearches(): QuickSearchEntry[] {
+    if (!this.config) {
+      return [];
+    }
+
+    const entries: QuickSearchEntry[] = this.config.quickSearches.map(
+      entry => new QuickSearchEntry(entry),
+    );
+
+    return entries;
+  }
+
+  private updateSearchTerm(e: CustomEvent): void {
+    this.searchTerm = e.detail.value;
+  }
+
+  private searchCleared(): void {
+    this.searchTerm = '';
+    this.emitSearchClearedEvent();
+    /* istanbul ignore else */
+    if (this.transcriptView) {
+      this.transcriptView.selectedSearchResultIndex = 0;
+    }
+    /* istanbul ignore else */
+    if (this.searchResultsSwitcher) {
+      this.searchResultsSwitcher.currentResultIndex = 0;
+    }
+  }
+
+  private emitSearchClearedEvent(): void {
+    const event = new Event('searchCleared');
+    this.dispatchEvent(event);
+  }
+
+  private searchResultIndexChanged(e: CustomEvent): void {
+    if (!this.transcriptView) {
+      return;
+    }
+    this.transcriptView.selectedSearchResultIndex = e.detail.searchResultIndex;
+    this.transcriptView.scrollToSelectedSearchResult();
+  }
+
+  private searchEnterKeyPressed(e: CustomEvent): void {
+    const event = new CustomEvent('searchRequested', {
+      detail: { searchTerm: e.detail.value },
+    });
+    this.dispatchEvent(event);
+  }
+
+  private get transcriptEntries(): TranscriptEntryConfig[] {
+    return this.transcriptConfig ? this.transcriptConfig.entries : [];
+  }
+
+  private get transcriptView(): TranscriptView | null {
+    /* istanbul ignore next */
+    return this.shadowRoot
+      ? (this.shadowRoot.querySelector('transcript-view') as TranscriptView)
+      : null;
+  }
+
+  private get audioElement(): AudioElement | null {
+    /* istanbul ignore next */
+    return this.shadowRoot
+      ? (this.shadowRoot.querySelector('audio-element') as AudioElement)
+      : null;
+  }
+
+  private get playbackControls(): PlaybackControls | null {
+    /* istanbul ignore next */
+    return this.shadowRoot
+      ? (this.shadowRoot.querySelector('playback-controls') as PlaybackControls)
+      : null;
+  }
+
+  private get searchResultsSwitcher(): SearchResultsSwitcher | null {
+    /* istanbul ignore next */
+    return this.shadowRoot
+      ? (this.shadowRoot.querySelector('search-results-switcher') as SearchResultsSwitcher)
+      : null;
+  }
+
+  private changePlaybackRate(e: CustomEvent): void {
+    this.playbackRate = e.detail.playbackRate;
```

Comment:
(maybe it's my JS ptsd,  but object traversal without an error handler makes me hyperventilate)

May be better to use object destructure to ensure that there is a value.
Like: 
```
const { detail } = e;
const { playbackRate = 1 } = detail;
```

Reply:
Check em now. I updated all of the spots where we're using CustomEvents to handle failures better and added a bunch o' tests for those cases.